board.rb

```diff
@@ -40,7 +40,7 @@ def initialize
   end
 
   def set_board(board_string)
-    grid = board_string.strip().split("\n").map{|row| row.strip().split('')}
+    grid = board_string.gsub(' ', '').strip().split("\n").map{|row| row.strip().split('')}
```

Comment:
`gsub` makes the parsing robust against spaces between the pieces. 

This allows parsing boards that are saved ike this:
```
rnbqkbnr
pppppppp
________
________
________
________
PPPPPPPP
RNBQKBNR
white to move
```

As well as boards that are saved like this with spacing between the pieces:
```
 ♜  ♞  ♝  ♛  ♚  ♝  ♞  ♜ 
 ♟  ♟  ♟  ♟  ♟  ♟  ♟  ♟ 
 _  _  _  _  _  _  _  _ 
 _  _  _  _  _  _  _  _ 
 _  _  _  _  _  _  _  _ 
 _  _  _  _  _  _  _  _ 
 ♙  ♙  ♙  ♙  ♙  ♙  ♙  ♙ 
 ♖  ♘  ♗  ♕  ♔  ♗  ♘  ♖ 
white to move.
```

I figured it would be better to make the parsing more permissive than making the file format restrictive.

---

From my understanding, `gsub` replaces _all_ occurences of the pattern, whereas `sub` replaces only the first occurence. So because there are multiple spaces in the board, I need to replace them all.

---

Thanks for the suggestion about the syntax, I've cleaned it up.